### ISSUE TITLE
Report if 'Assert.Fail' is within if block

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#2179]
We should report if `Assert.Fail` is invoked within an `if` block and suggest to replace it with `Assert.That(..., Is.False)` or `Assert.That(..., Is.True)`

Example:
Following code should be reported:
```c#
if (!list.Any())
{
    Assert.Fail("missing entries");
}
```
And fixed into:
```c#
Assert.That(list.Any(), Is.True, "missing entries");
```

Same applies for:
```c#
if (list.Any())
{
    Assert.Fail("contains entries");
}
```

It should be reported and fixed into:
```c#
Assert.That(list.Any(), Is.False, "contains entries");
```
